fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreBase.java

```diff
@@ -218,6 +219,15 @@ default FDBStoreTimer getTimer() {
         default IndexState needRebuildIndex(Index index, long recordCount, boolean indexOnNewRecordTypes) {
             return FDBRecordStore.disabledIfTooManyRecordsForRebuild(recordCount, indexOnNewRecordTypes);
         }
+
+        @Nonnull
+        default CompletableFuture<IndexState> needRebuildIndex(Index index,
```

Comment:
Okay, added some javadoc explaining what this function does. I also specified that the return future should complete after any accessed futures, though I think the only consequence would be some logging, which is probably enough to justify mentioning it, though I don't think it would be the end of the world if not implemented correctly.